### PR TITLE
Reusable Box component

### PR DESCRIPTION
**Pull request by AhmedAdel999**
- Created `Box` component.
- Added test file and storybook file.
- Used it in widgets.

### CODE REVIEW

File: modules/shared/components/atoms/Box/IBox.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import type { ReactElement } from 'react';
+
+declare namespace IBox {
+  export interface IProps {
+    classes?: string;
```
Comment (omarawad0) [inline modules/shared/components/atoms/Box/IBox.d.ts:5]: What is the use for the Classes prop?

Comment (AhmedAdel999) [reply modules/shared/components/atoms/Box/IBox.d.ts:5]: In case we wanted to apply additional classes.

**Review by omarawad0 — APPROVED** [review]
LGTM

File: modules/shared/components/atoms/Box/IBox.d.ts (hunk 1 above)
Comment (Nilegfx) [reply modules/shared/components/atoms/Box/IBox.d.ts:5]: do you test this className prop? do we use it at the moment? if the answer is No, then probably we should remove it and introduce it when we have the first use case

**Pull request merged**